cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/set_asset_claimer.rs

```diff
@@ -0,0 +1,257 @@
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = ParaToParaThroughAHTest::new(test_args);
+	execute_test(test.clone(), bob_location.clone(), transfer_assets);
+
+	let alice_assets_after = query_balance(&alice_account, &RelayLocation::get());
+	assert_eq!(alice_assets_after, amount_to_send);
+
+	let test_args = TestContext {
+		sender: bob_account.clone(),
+		receiver: alice_account.clone(),
+		args: TestArgs::new_para(
+			alice_location.clone(),
+			alice_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = ParaToParaThroughAHTest::new(test_args);
+	execute_test(test.clone(), bob_location.clone(), claim_assets);
+
+	let bob_assets_after = query_balance(&bob_account, &RelayLocation::get());
+	assert_eq!(bob_assets_after, amount_to_send);
+}
+
+fn account_and_location(account: &str) -> (AccountId32, Location) {
+	let account_id = PenpalA::account_id_of(account);
+	let account_clone = account_id.clone();
+	let location: Location =
+		[Junction::AccountId32 { network: Some(Westend), id: account_id.into() }].into();
+	(account_clone, location)
+}
+
+fn fund_account(account: &AccountId, amount: u128) {
+	let asset_owner = PenpalAssetOwner::get();
+	PenpalA::mint_foreign_asset(
+		<PenpalA as Chain>::RuntimeOrigin::signed(asset_owner),
+		Location::parent(),
+		account.clone(),
+		amount,
+	);
+}
+
+fn query_balance(account: &AccountId, asset_location: &Location) -> u128 {
+	PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(asset_location.clone(), account)
+	})
+}
+
+fn execute_test(
+	test: ParaToParaThroughAHTest,
+	claimer: Location,
+	xcm_fn: impl Fn(ParaToParaThroughAHTest, Location) -> <PenpalA as Chain>::RuntimeCall,
+) {
+	let call = xcm_fn(test.clone(), claimer.clone());
+	PenpalA::execute_with(|| {
+		assert!(call.dispatch(test.signed_origin).is_ok());
+	});
+}
+
+fn transfer_assets(
+	test: ParaToParaThroughAHTest,
+	claimer: Location,
+) -> <PenpalA as Chain>::RuntimeCall {
+	type RuntimeCall = <PenpalA as Chain>::RuntimeCall;
+
+	let local_xcm = Xcm::<RuntimeCall>::builder_unsafe()
+		.set_asset_claimer(claimer.clone())
+		.withdraw_asset(test.args.assets.clone())
+		.clear_origin()
+		.build();
+
+	RuntimeCall::PolkadotXcm(pallet_xcm::Call::execute {
+		message: bx!(VersionedXcm::from(local_xcm)),
+		max_weight: Weight::from_parts(4_000_000_000_000, 300_000),
+	})
+}
+
+fn claim_assets(
+	test: ParaToParaThroughAHTest,
+	claimer: Location,
+) -> <PenpalA as Chain>::RuntimeCall {
+	type RuntimeCall = <PenpalA as Chain>::RuntimeCall;
+
+	let local_xcm = Xcm::<RuntimeCall>::builder_unsafe()
+		.claim_asset(test.args.assets.clone(), Here)
+		.deposit_asset(AllCounted(test.args.assets.len() as u32), claimer)
+		.build();
+
+	RuntimeCall::PolkadotXcm(pallet_xcm::Call::execute {
+		message: bx!(VersionedXcm::from(local_xcm)),
+		max_weight: Weight::from_parts(4_000_000_000_000, 300_000),
+	})
+}
```

Comment:
same here, called just from one place, drop this fn and move the inner code at the call site

Reply:
dropped